lua/starfall/libs_sh/vr.lua

```diff
@@ -0,0 +1,318 @@
+	getply = instance.Types.Player.GetPlayer
+end)
+
+--- Checks wether the player is in VR
+-- @param target player to check
+-- @return boolean true if player is in VR
+function vr_library.isPlayerInVR(ply)
+	return vrmod.IsPlayerInVR(getply(ply))
+end
+
+--- Checks wether the player is using empty hands
+-- @param target player to check
+-- @return boolean true if player is using empty hands
+function vr_library.usingEmptyHands(ply)
+	return vrmod.UsingEmptyHands(getply(ply))
+end
+
+--HMD
+
+--- returns the HMD position
+-- @param target player to get the HMD position from
+-- @return vector position
+function vr_library.getHMDPos(ply)
+	return vwrap(vrmod.GetHMDPos(getply(ply)))
+end
+
+--- returns the HMD angles
+-- @param target player to get the HMD angles from
+-- @return angle angles
+function vr_library.getHMDAng(ply)
+	return awrap(vrmod.GetHMDAng(getply(ply)))
+end
+
+--- returns the HMD pose
+-- @param target player to get the HMD pose from
+-- @return vector position 
+-- @return angle angles
+function vr_library.getHMDPose(ply)
+	local pos, ang = vrmod.GetHMDPose(getply(ply))
+	return vwrap(pos), awrap(ang)
+end
+
+--Left Hand
+
+--- returns the left hand position
+-- @param target player to get the left hand position from
+-- @return vector position
+function vr_library.getLeftHandPos(ply)
+	return vwrap(vrmod.GetLeftHandPos(getply(ply)))
+end
+
+--- returns the left hand angles
+-- @param target player to get the left hand angles from
+-- @return angle angles
+function vr_library.getLeftHandAng(ply)
+	return awrap(vrmod.GetLeftHandAng(getply(ply)))
+end
+
+--- returns the left hand pose
+-- @param target player to get the left hand pose from
+-- @return vector position 
+-- @return angle angles
+function vr_library.getLeftHandPose(ply)
+	local pos, ang = vrmod.GetLeftHandPose(getply(ply))
+	return vwrap(pos), awrap(ang)
+end
+
+--Right Hand
+
+--- returns the right hand position
+-- @param target player to get the right hand position from
+-- @return vector position
+function vr_library.getRightHandPos(ply)
+	return vwrap(vrmod.GetRightHandPos(getply(ply)))
+end
+
+--- returns the left hand angles
+-- @param target player to get the right hand angles from
+-- @return angle angles
+function vr_library.getRightHandAng(ply)
+	return awrap(vrmod.GetRightHandAng(getply(ply)))
+end
+
+--- returns the left hand pose
+-- @param target player to get the right hand pose from
+-- @return vector position 
+-- @return angle angles
+function vr_library.getRightHandPose(ply)
+	local pos, ang = vrmod.GetRightHandPose(getply(ply))
+	return vwrap(pos), awrap(ang)
+end
+
+if CLIENT then
+
+
+
+--- returns the a controller's input state, may return boolean, number or vector.
+-- @param actionname to check control of, check VR enums
+-- @return boolean, vector or number of input
+-- @client
+function vr_library.getInput(actionname)
+	checkluatype(actionname, TYPE_STRING) 
+	local var = vrmod.GetInput(actionname)
```

Comment:
Sorry to make you change this again, but on second thought, `instance.WrapObject(vrmod.GetInput(actionname))` would be best here. Everything else looks fine.

Reply:
Are you sure? this function could be called quite often (per frame, per tick), having it optimized for those three types should be good then, right? Unless WrapObject does it faster

Reply:
@Neatro

Reply:
Ah, ok. Then lets leave it.